packages/expo-cli/src/commands/run/index.ts

```diff
@@ -0,0 +1,53 @@
+import { Command } from 'commander';
+
+import CommandError from '../../CommandError';
+import prompt from '../../prompts';
+import buildAndroidClientAsync from './buildAndroidClientAsync';
+
+type Options = {
+  platform?: string;
+  buildVariant?: string;
+};
+
+async function runAsync(projectRoot: string, options: Options) {
+  const platform = options.platform?.toLowerCase() ?? (await promptForPlatformAsync());
+  if (platform === 'android') {
+    if (typeof options.buildVariant !== 'string') {
+      throw new CommandError('--build-variant must be a string');
+    }
+    await buildAndroidClientAsync(projectRoot, {
+      buildVariant: options.buildVariant,
+    });
+  } else {
+    throw new Error('platform not implemented');
+  }
+}
+
+async function promptForPlatformAsync(): Promise<'android' | 'ios'> {
+  const { platform } = await prompt({
+    type: 'select',
+    message: 'Choose platform',
+    name: 'platform',
+    choices: [
+      {
+        title: 'Android',
+        value: 'android',
+      },
+      {
+        title: 'iOS',
+        value: 'ios',
+      },
+    ],
+  });
+  return platform;
+}
+
+export default function (program: Command) {
+  program
+    .command('run [path]')
+    .helpGroup('experimental')
+    .description('Build a development client and run it in on a device.')
+    .option('-p --platform <platform>', 'Platform: [android|ios]', /^(android|ios)$/i)
+    .option('--build-variant [name]', '(Android) build variant', 'release')
```

Comment:
hmm if this is a cross platform command, then we should probably attempt to make the arguments cross-platform too. If that's not feasible, then we should split ios and android build commands up.

Reply:
The intent was to match the convention used by `eas build`, where the same command builds iOS and Android apps: `eas build --platform ios`, `eas build --platform android` or even `eas build --platform all`.

It would be possible to do `expo run:ios` and `expo run:android`, but this (specifying the platform in the command name) is a pattern we've been trying to avoid in new code.

Cross-plaform arguments: Are you suggesting a new argument that maps to build variant name on Android and build configuration name on iOS?

Reply:
Renamed to `run:android` for now.  ¯\\\_(ツ)\_/¯